homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -65,6 +66,17 @@ async def async_setup_gateway_entry(
         sw_version=gateway_info.firmware_version,
     )
 
+    for subdevice in subdevices:
+        device_registry.async_get_or_create(
```

Comment:
Please remove this and just implement `device_info` in the entity class instead.

Reply:
In this case that would not be the preferred way because the xiomi_miio library currently can discover 51 different zigbee sub devices where we know the zigbee_model, model and device name. However only 2 of those are actually implemented as HomeAssistant entities (SensorHT and AqaraHT). There are a few more +- 6 devices where all control methods are already implemented but not added as HomeAssistant entities yet.

For the rest of the 43 known and discoverable devices, we are looking for people that actually own those devices such that we can check and implement the device specific get/set functions of there properties to implement the entities.

Personally I would like those 49 device types to be shown in the device overview without entities but with all proper information like firmware_version, name, model, sid etc. So that users can see that those devices are discovered and view there information and know that these devices could be supported by HomeAssistant with a relatively small effort (all infrastructure and general methods to control those devices is already there, basically we only need to know the names of the properties like "channel_0", "load_power", "neutral_0", "neutral_1", "channel_1", "temperature", "humidity", "pressure", "power" etc.)

That would not be possible if the device_info is set in the entities

Reply:
I don't think we should add devices that are not providing entities or is a hub type device.

Reply:
Another point is that some devices like buttons and remotes will only fire events and will probably not have enties.
Schould those devices also not be registered?

Reply:
I personnaly like to have an overvieuw of my devices so I can easily look up models etc. but if you insist I will remove the devices from the registry and only provide them for devices that have entities.

Reply:
We should only add devices for things that expose features to home assistant.

Reply:
I have moved the device_info to the entitie